src/app/state/todos/todos.selectors.ts

```diff
@@ -1,3 +1,17 @@
+import { createSelector } from '@ngrx/store';
+
 export function getTodos(state) {
-  return state.todos;
+  return state.todos.todos;
 }
+
+export function getSelectedTodoId(state) {
+  return state.todos.selectedTodoId;
+}
+
+export const getSelectedTodo = createSelector(
+  getTodos,
+  getSelectedTodoId,
+  (todos, selectedTodoId) => {
+    return todos.filter(todo => todo.id == selectedTodoId)[0];
+  }
+);
```

Comment:
Now the way we can actually select our todo, is by composing the `getTodos` and getSelectedTodoId` selectors together.

By keeping our selectors pure functions, we're able to compose them together to select (and map out) more meaningful data from our state